models/ctsm/R/write_params-clm.R

```diff
@@ -1,670 +1,101 @@
-      #  ncdf4::ncvar_put(nc=fates.param.nc, varid='minspread',vals=pft[v])
-      #}
-      
-      ## LITTERCLASS indexed (Size:6)
-      ## MCD: skipping for now until there's demonstrated demand because it requires expanding every variable out into VARNAME_[1..n]
-      # low_moisture_C	Intercept (constant) of fuel moisture to burned fraction term for drier fuel	litterclass	
-      # low_moisture_S	Slope of fuel moisture to burned fraction term for drier fuel	litterclass	
-      # max_decomp	Maximum decomposition rate of litter in the absence of moisture or temperature stress, per fuel class	litterclass	y-1
-      # mid_moisture	Parameter of burned fraction term. Below this 'low' constants apply, above this, 'mid' constants apply, 	litterclass	
-      # mid_moisture_C	Intercept (constant) of fuel moisture to burned fraction term for wetter fuel	litterclass	
-      # min_moisture	Parameter of burned fraction term. Below this value all litter is burned by a fire. Above, 'low' constants apply	litterclass	
-      # FBD	Fuel Bulk Density of fuel class 	litterclass	kg m-3
-      # alpha_FMC	Parameter of function relating fuel moisture content to meteorological fire danger index 	litterclass	
-      # SAV	Surface Area to Volume Ratio of fuel class 	litterclass	cm-1
-      
-      ## NCWD dimensioned       Size:4
-      if(var == "CWD_frac1"){            ##Fraction of coarse woody debris (CWD) that is moved into each of the four woody fuel classes
-        ncdf4::ncvar_put(nc=fates.param.nc, varid='fates_CWD_frac', start = 1, count = 1,
-                         vals=pft[v])
-      }
-      if(var == "CWD_frac2"){
-        ncdf4::ncvar_put(nc=fates.param.nc, varid='fates_CWD_frac', start = 2, count = 1,
-                         vals=pft[v])
-      }
-      if(var == "CWD_frac3"){
-        ncdf4::ncvar_put(nc=fates.param.nc, varid='fates_CWD_frac', start = 3, count = 1,
-                         vals=pft[v])
-      }
-      if(var == "CWD_frac4"){
-        ncdf4::ncvar_put(nc=fates.param.nc, varid='fates_CWD_frac', start = 4, count = 1,
-                         vals=pft[v])
-      }
-      
-      
-    } ## end loop over VARIABLES
-  } ## end loop over PFTs
-  #ncdf4::nc_close(param.nc)
-  ncdf4::nc_close(clm.param.nc)
+#' /dontrun{
+#' trait.values <- list(c4_grass = list(sla = 0.03, c2n_leaf = 35, stom_slope = 10, fineroot2leaf = 2, vcmax = 50))
+#' write_params_ctsm(trait.values = trait.values, run.id = 1)
+#' }
 
-  
-}
+# TODO: update this to read param files (CTSM and FATES) out of the refcase directory, not the PEcAn package
+# TODO: update to allow it to pick between CLM4.5 and CLM5 (CLM5 == CTSM) parameter set based on refcase, user selection
+# TODO: match Pecan pft names with CTSM pft names when Pecan name is not identical to a CTSM name
+## See issue https://github.com/PecanProject/pecan/issues/1008
 
+write_params_ctsm <-
+  function(defaults = system.file('clm5_params.c171117_0001.nc', package = 'PEcAn.CTSM'),
```

Comment:
I would instead advocate that we use a refrun script, like the one I wrote for the Docker, that places the clm param file in the refrun directory such that we have the correct version in the refrun directory. Then we grab the default file from that directory instead of the pecan package directory

Reply:
example: https://github.com/serbinsh/ctsm_containers/blob/master/ctsm_run_scripts/create_case_ctsmfates_1pt_example_PA-SLZ.sh

where we would have something like:

export CLM_PARAM_FILE_PATH={path to cesm datasets/lnd/clm2/}
export CLM_PARAM_FILE=clm5_params.c171117_0001.nc /new/run/dir

Reply:
@serbinsh I don't think I quite understand this. I've looked over your example, but that is part of creating a case for a CLM run, which is not what we want to do here? 

So we want to have a shell script with the two `export` lines (in models/ctsm?). This will indicate where the initial CLM parameter file is, which will not be in the Pecan repository but will only exists as a host machine file? Will it be an "inputs"-type file like in Bety? And will that shell script then be run by a Dockerfile? Which one?

Reply:
@KristinaRiemer please move shawn's suggestions to new PR and then this can be merged